app/controllers/article_controller.rb

```diff
@@ -208,4 +214,13 @@ def eds_authenticated_user?
   def has_search_parameters?
     params[:q].present? || params[:f].present?
   end
+
+  def extract_fulltext_link(document, type)
+    links = document.fetch('eds_fulltext_links', [])
+    links.each do |link|
+      next if link['url'].blank? || link['url'] == 'detail'
+      return link['url'] if link['type'] == type.to_s
+    end
+    raise ArgumentError, "Missing #{type} fulltext link in document #{document.id}"
```

Comment:
Yes, this failure is a 500 error. This failure happens when the URL is blank or "detail" (as returned by the EDS API) or when the specified type is wrong (e.g., `/article/123/some-random-type/fulltext`). Maybe it's really a 404 case?